File: FPG.Core/Factories/CroppingAlgorithmFactory.cs
using System;

using FPG.Algorithms;
using FPG.Interfaces;

namespace FPG.Factories
{
    public static class CroppingAlgorithmFactory
    {
        public static ICroppingAlgorithm Create(String name)
        {
            switch (name.ToLower())
            {
                case "even":
                    return new EvenCroppingAlgorithm();
                default:
                    throw new ArgumentException("Unknown algorithm name");
            };
        }
    }
}

File: src/lib/core/factories/CroppingAlgorithmFactory.dart
import 'package:fpg/core/algorithms/cropping/EvenCroppingAlgorithm.dart';
import 'package:fpg/core/interfaces/ICroppingAlgorithm.dart';

class CroppingAlgorithmFactory {
  static ICroppingAlgorithm create(String name) {
    switch (name.toLowerCase()) {
      case "even":
        return EvenCroppingAlgorithm();
      default:
        throw FormatException("Unknown algorithm name");
    }
  }
}
